### DISCUSSION TITLE
error during installation

### DISCUSSION THREAD

**Discussion by rabol** [GeorgeFourkas/ApiAutoPilot#9] [Q&A]
Hi

on a fresh laravel installation I get this:

```
 test php artisan apiautopilot:install

   ErrorException 

  file_get_contents(/Users/rabol/code/web/test/app/Http\Kernel.php): Failed to open stream: No such file or directory

```

Any hints ?

**Comment by GeorgeFourkas** [comment]
Hello and thank you for using Api Auto Pilot, about the error that shows
when you are running the installation command, I ve noticed a 'test'
keyword before php. If you want to test the package there are already
written and passing tests. Are you trying to test the package or  use it in
a real use case?

About the GitHub issues, I will make sure to enable them as soon as
possible,


Στις Κυρ, 11 Δεκ 2022, 15:56 ο χρήστης Steen Rabol ***@***.***>
έγραψε:

**Reply by rabol** [reply]
Well... yes I'm trying to test the package, and my laravel app is in a folder called test, which is why test is part of the path

Here is what I have done:

in my 'web' folder (using Laravel Valet)

1) Laravel new test
2) composer require laravel/breeze
3) php artisan breeze:install --pest
4) composer require apiautopilot/apiautopilot
5) php artisan apiautopilot:install

and it's step 5 that issue the error

**Reply by GeorgeFourkas** [reply]
It seems like a middleware registration issue, well to fix the issue manually you should add 
```
'modelSearch' =>  \ApiAutoPilot\AutoPilotApi\Http\Middleware\ModelSearch::class
```
to the
 ```
protected $routeMiddleware
```
array in the app/Http/Kernel.php file

**Comment by rabol** [comment]
In the install command file line 32
change it to this:
```
 $kernelContent = file_get_contents(app_path('Http/Kernel.php'));
```

note the forward slash - looks like you are using windows :)

**Comment by rabol** [comment]
also, the publishConfig() does not publish the config
